config/validator_test.go

```diff
@@ -258,11 +246,106 @@ func TestValidatorShouldFailWithMissingClusterDefinition(t *testing.T) {
 	var size httphandlerconfig.HumanSizeUnits
 	size.SizeInBytes = 2048
 	regions := map[string]regionsconfig.Region{"testregion": regionConfig}
-	yamlConfig := PrepareYamlConfig(size, 31, 45, "127.0.0.1:81", "127.0.0.1:1234", "127.0.0.1:1235", regions)
+	yamlConfig := PrepareYamlConfig(size, 31, 45, "127.0.0.1:81",
+		"127.0.0.1:1234", "127.0.0.1:1235", regions, nil)
 	valid, validationErrors := yamlConfig.RegionsEntryLogicalValidator()
 	assert.False(t, valid)
 	assert.Equal(
 		t,
 		errors.New("No clusters defined for region \"testregion\""),
 		validationErrors["RegionsEntryLogicalValidator"][0])
 }
+
+func TestValidatorShouldFailWithEmptyTransportsDefinition(t *testing.T) {
+	transports := make(transportconfig.Transports, 0)
+	var size httphandlerconfig.HumanSizeUnits
+	size.SizeInBytes = 2048
+	yamlConfig := PrepareYamlConfig(size, 31, 45, "127.0.0.1:81",
+		"127.0.0.1:1234", "127.0.0.1:1235", nil, transports)
+	valid, validationErrors := yamlConfig.TransportsEntryLogicalValidator()
+	assert.False(t, valid)
+	assert.Equal(
+		t,
+		errors.New("Empty transports definition"),
+		validationErrors["TransportsEntryLogicalValidator"][0])
+}
+
+func TestValidatorShouldProcessTransportsWithSuccess(t *testing.T) {
+	validTransports := transportconfig.Transports{
+		transportconfig.Transport{
+			Name: "TestTransport",
+			Matchers: transportconfig.ClientTransportMatchers{
+				Method:     "GET",
+				Path:       ".*",
+				QueryParam: "",
+			},
+		},
+	}
+	var size httphandlerconfig.HumanSizeUnits
+	size.SizeInBytes = 2048
+	yamlConfig := PrepareYamlConfig(size, 31, 45, "127.0.0.1:81",
+		"127.0.0.1:1234", "127.0.0.1:1235", nil, validTransports)
+	valid, _ := yamlConfig.TransportsEntryLogicalValidator()
+	assert.True(t, valid)
+}
+
+func TestValidatorShouldProcessTransportsWithSuccessWithNotDefinedMatchersProperties(t *testing.T) {
+	invalidTransports := transportconfig.Transports{
+		transportconfig.Transport{
+			Name: "TestTransport",
+			Matchers: transportconfig.ClientTransportMatchers{
+				Method: "GET",
+			},
+		},
+	}
+	var size httphandlerconfig.HumanSizeUnits
+	size.SizeInBytes = 2048
+	yamlConfig := PrepareYamlConfig(size, 31, 45, "127.0.0.1:81",
+		"127.0.0.1:1234", "127.0.0.1:1235", nil, invalidTransports)
+	valid, _ := yamlConfig.TransportsEntryLogicalValidator()
+	assert.True(t, valid)
+}
+
+func TestShouldValidWithEmptyPropertiesInmatchersDefinition(t *testing.T) {
+	invalidTransports := transportconfig.Transports{
+		transportconfig.Transport{
+			Name: "TestTransport123",
```

Comment:
Maybe the transport's name should be something like `CatchAllRequestsTransport` so people are aware of such a funcrionality